indexbuffer.cpp

```diff
@@ -0,0 +1,35 @@
+#include "indexbuffer.h"
+
+#include "renderutils.h"
+
+// Ensure unsigned int is the same as a GLuint
+static_assert(sizeof(unsigned int) == sizeof(GLuint));
+
+IndexBuffer::IndexBuffer(const unsigned int* data, unsigned int count)
+    : mCount(0)
+{
+    GLCall(glGenBuffers(1, &mRendererID));
+    GLCall(glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, mRendererID));
+    GLCall(glBufferData(GL_ELEMENT_ARRAY_BUFFER, count * sizeof(unsigned int), data, GL_STATIC_DRAW));
+    mCount = count;
+}
+
+IndexBuffer::~IndexBuffer()
+{
+    GLCall(glDeleteBuffers(1, &mRendererID));
+}
```

Comment:
If you need an explicit destructor (which it appears you do) you should define or explicitly delete the copy and move constructors. Remember the rule of 5 :)